pkg/stanza/fileconsumer/file.go

```diff
@@ -32,28 +33,15 @@ type Manager struct {
 	maxBatches    int
 	maxBatchFiles int
 
-	previousPollFiles []*reader.Reader
-	knownFiles        []*reader.Metadata
-
-	// This value approximates the expected number of files which we will find in a single poll cycle.
-	// It is updated each poll cycle using a simple moving average calculation which assigns 20% weight
-	// to the most recent poll cycle.
-	// It is used to regulate the size of knownFiles. The goal is to allow knownFiles
-	// to contain checkpoints from a few previous poll cycles, but not grow unbounded.
-	movingAverageMatches int
+	activeFiles       *fileset.Fileset[*reader.Reader]
```

Comment:
Can we call this `currentPollFiles`?

Reply:
Related thought - we could consolidate "current" and "previous" into "open" by making it `[2]*fileset.Fileset[*reader.Reader]`. Maybe best to keep this in mind as a possible refactoring after this PR.

Reply:
Sure.